File: Java/src/main/java/edu/bu/filehandler/LogFileHandler.java
package edu.bu.filehandler;

import java.io.BufferedWriter;
import java.io.FileOutputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;

public class LogFileHandler {

    private String fileName;

    public LogFileHandler() {
    }

    public LogFileHandler(String fileName) {
        this.fileName = fileName;
    }

    public void addLog(boolean read,int serializationType, boolean seq,String datatype, double iotime, double totaltime,boolean taskset){

        String log;
        if (read)
            log="[ReadTimeJAVA]#"+taskset+"#";
        else
            log="[WriteTimeJAVA]#"+taskset+"#";
        String method="";
        switch (serializationType) {
            case 1:
                method = "Java Default";
                break;
            case 2:
                method = "Java Json+Gzip";
                break;
            case 3:
                method = "Java Bson";
                break;
            case 4:
                method = "Java ProtoBuf";
                break;
            case 5:
                method = "Java Kryo";
                break;
            case 6:
                method = "Java Byte Buffer";
                break;
            case 7:
                method = "Java FlatBuffers";
                break;
            case 8:
                method = "Java Json";
                break;
            case 9:
                method = "Java Gson";
                break;
        }
        String seqString="true";
        if (!seq)
            seqString="false";

        log+=method+"#";
        log+=seqString+"#";
        log+=datatype+"#";
        log+=iotime+"#";
        log+=totaltime+"\n";
        try (Writer writer = new BufferedWriter(new OutputStreamWriter(
                new FileOutputStream(this.fileName,true), "utf-8"))) {
            writer.write(log);
        }catch (Exception ex) {

        }
    }
}

File: C++/src/filehandler/source/LogFileHandler.cpp
//
// Created by saeed on 12/19/19.
//

#include "LogFileHandler.h"

LogFileHandler::LogFileHandler() {}

LogFileHandler::~LogFileHandler() {

}

LogFileHandler::LogFileHandler(string fileName) {
    logFile.open (fileName, ios::out | ios::app );
}

void LogFileHandler::addLog(string log) {

    logFile << log<<"\n";
}

void LogFileHandler::addLog(bool read,int serializationType, bool seq, string datatype, double iotime, double totaltime, bool taskset) {

    if (read) {
        logFile << "[ReadTimeCPP]#";
    } else
        logFile << "[WriteTimeCPP]#";

    if (taskset){
        logFile<<"true#";
    } else
        logFile<<"false#";

    string method="";
    switch (serializationType){
        case 1:
            method="C++ HandCoded";
            break;
        case 2:
            method="C++ inPlace";
            break;
        case 3:
            method="C++ Boost";
            break;
        case 4:
            method="C++ ProtoBuf";
            break;
        case 5:
            method="C++ Bson";
            break;
        case 6:
            method="C++ FlatBuffers";
            break;

    }
    string seqString="true";
    if (!seq)
        seqString="false";

    logFile<<method<<"#";
    logFile<<seqString<<"#";
    logFile<<datatype<<"#";
    logFile<<iotime<<"#";
    logFile<<totaltime<<"\n";
}

void LogFileHandler::flushLogFile() {
    logFile.close();
}
